### DISCUSSION TITLE
Sorting tabs that are not active

### DISCUSSION THREAD

**Discussion by dawsers** [sxyazi/yazi#1835] [1. Q&A] [answered]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

kitty 0.36.4

### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (e8c1d62 2024-10-24)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (Arch Linux 2024-09-05)

Emulator
    Emulator.via_env: ("xterm-kitty", "")
    Emulator.via_csi: Ok(Kitty)
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kgp

Desktop
    XDG_SESSION_TYPE           : Some("wayland")
    WAYLAND_DISPLAY            : Some("wayland-1")
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: Some("4520b30d498daca8079365bdb909a8dea38e8d55_1729846369_2041814869")
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/usr/bin/bash")
    EDITOR             : Some("/usr/bin/nvim")
    VISUAL             : Some("/usr/bin/nvim")
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default     : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR (block)", for_: None, spread: true })
    block-create: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR (block)", for_: None, spread: true })
    block-rename: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR (block)", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : No such file or directory (os error 2)
    tmux build flags   : enable-sixel=Unknown
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.3
    magick           : 7.1.1-39
    fzf              : 0.55.0
    fd               : 10.2.0
    rg               : 14.1.1
    chafa            : 1.14.2
    zoxide           : 0.9.6
    7z               : 17.05
    7zz              : No such file or directory (os error 2)
    jq               : 1.7.1
```

### Describe the question

Is there any way to keep the sorting on tabs that are no longer active?

I have a plugin that creates a dual pane configuration, [dual-pane.yazi](https://github.com/dawsers/dual-pane.yazi), and the tab shown on the non-active pane loses its sorting in certain cases, like for example when adding a new tab.

Use `mtime` sorting, for example. Open dual-pane. Both panes will show the same directory. Now create a new tab. The new tab will be shown on the active pane, and it will be sorted correctly. The other pane, which was showing the correct sorting, now loses it.

Of course, it is not possible to reproduce this in standard Yazi, as it only shows the active tab, which is always correctly sorted. 

My guess is there is probably no sorting cache, and sorting is discarded when a tab is no longer active. So the other pane, which now contains an inactive tab, isn't sorted.

Is there a way to force sorting on a tab and keeping it sorted even when it is not active?

### Validations

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment, marked as answer]
> the tab shown on the non-active pane loses its sorting in certain cases

Done in https://github.com/sxyazi/yazi/issues/1852 and https://github.com/sxyazi/yazi/issues/1981
